### DISCUSSION TITLE
Add whitelist and blacklist only to host names and subdomains from supplied text file. Total circumvention will broke online-banking hosts.

### DISCUSSION THREAD

**Discussion by babitsyn** [hufrea/byedpi#221] [Ideas]
It would be very useful option sometimes, for example to visit banking, government and etc. sites.

**Comment by zimbabwe303** [comment]
You already can do that using the `--hosts` (-H) option:

`--hosts whitelist.txt -An --hosts blacklist.txt [spoofing params]`

Of course using both whitelist and blacklist at the same time is redundant (see examples below for correct usage). The whilelist and blacklist should be plain text files with domain names listed one at a line. For blacklist, if you have several `--auto` (-A) parameters you should supply `--hosts blacklist.txt` every time.

Whitelist example:
`ciadpi -H whitelist.txt -An -s1 -q1 -Y -At -T2 -b1000 -S -f-1 -r1+sm -As`

Blacklist example:
`ciadpi -H blacklist.txt -s1 -q1 -Y -At -H blacklist.txt -T2 -b1000 -S -f-1 -r1+sm -As`

**Reply by celomu** [reply]
How does the app know where this white or black list is located on the phone?
